src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/scatter_nd_update_ref.cl

```diff
@@ -49,81 +68,123 @@ KERNEL(scatter_nd_update_ref)(const __global INPUT0_TYPE* data,
+    #endif
 
-    for (int i = 0; i < size_to_update; i++) {
-        uint dst_idx = dst_offset + i;
-        uint up_idx = update_offset + i;
-        INPUT2_TYPE val = updates[up_idx];
+    #if INPUT0_DIMS == 4
+        const uint data_dim[INPUT0_DIMS] = {INPUT0_BATCH_NUM, INPUT0_FEATURE_NUM, INPUT0_SIZE_Y, INPUT0_SIZE_X};
+    #elif INPUT0_DIMS == 5
+        const uint data_dim[INPUT0_DIMS] = {INPUT0_BATCH_NUM, INPUT0_FEATURE_NUM, INPUT0_SIZE_Z, INPUT0_SIZE_Y, INPUT0_SIZE_X};
+    #elif INPUT0_DIMS == 6
+        const uint data_dim[INPUT0_DIMS] = {INPUT0_BATCH_NUM, INPUT0_FEATURE_NUM, INPUT0_SIZE_W, INPUT0_SIZE_Z, INPUT0_SIZE_Y, INPUT0_SIZE_X};
+    #endif
 
-    #if HAS_FUSED_OPS
-        #if OUTPUT_DIMS == 4
-            const uint y_pitch = OUTPUT_SIZE_X;
-            const uint f_pitch = y_pitch * OUTPUT_SIZE_Y;
-            const uint b_pitch = f_pitch * OUTPUT_FEATURE_NUM;
-
-            const uint b_remain = dst_idx % b_pitch;
-            const uint f_remain = b_remain % f_pitch;
-            const uint y_remain = f_remain % y_pitch;
-
-            const uint b = dst_idx / b_pitch;
-            const uint f = b_remain / f_pitch;
-            const uint y = f_remain / y_pitch;
-            const uint x = y_remain;
-        #elif OUTPUT_DIMS == 5
-            const uint y_pitch = OUTPUT_SIZE_X;
-            const uint z_pitch = y_pitch * OUTPUT_SIZE_Y;
-            const uint f_pitch = z_pitch * OUTPUT_SIZE_Z;
-            const uint b_pitch = f_pitch * OUTPUT_FEATURE_NUM;
-
-            const uint b_remain = dst_idx % b_pitch;
-            const uint f_remain = b_remain % f_pitch;
-            const uint z_remain = f_remain % z_pitch;
-            const uint y_remain = z_remain % y_pitch;
-
-            const uint b = dst_idx / b_pitch;
-            const uint f = b_remain / f_pitch;
-            const uint z = f_remain / z_pitch;
-            const uint y = z_remain / y_pitch;
-            const uint x = y_remain;
-        #elif OUTPUT_DIMS == 6
-            const uint y_pitch = OUTPUT_SIZE_X;
-            const uint z_pitch = y_pitch * OUTPUT_SIZE_Y;
-            const uint w_pitch = z_pitch * OUTPUT_SIZE_Z;
-            const uint f_pitch = w_pitch * OUTPUT_SIZE_W;
-            const uint b_pitch = f_pitch * OUTPUT_FEATURE_NUM;
-
-            const uint b_remain = dst_idx % b_pitch;
-            const uint f_remain = b_remain % f_pitch;
-            const uint w_remain = f_remain % w_pitch;
-            const uint z_remain = w_remain % z_pitch;
-            const uint y_remain = z_remain % y_pitch;
-
-            const uint b = dst_idx / b_pitch;
-            const uint f = b_remain / f_pitch;
-            const uint w = f_remain / w_pitch;
-            const uint z = w_remain / z_pitch;
-            const uint y = z_remain / y_pitch;
-            const uint x = y_remain;
+    // Get indices index
+    uint idx[INDICES_MAX_DIM] = {0};
+    uint rmd_idx = dim2;
+    for(int i = 0; i < INDICES_RANK - 1; ++i)
+    {
```

Comment:
Did you run cpplint on your local device?
I think it makes an error in cpplint like this

`{ should almost always be at the end of the previous line  [whitespace/braces] [4]`